packages/contracts-bedrock/src/L1/OptimismPortalInterop.sol

```diff
@@ -0,0 +1,819 @@
+        IAnchorStateRegistry newAnchorStateRegistry
+    );
+
+    /// @notice Thrown when a withdrawal has already been finalized.
+    error OptimismPortal_AlreadyFinalized();
+
+    /// @notice Thrown when the target of a withdrawal is unsafe.
+    error OptimismPortal_BadTarget();
+
+    /// @notice Thrown when the calldata for a deposit is too large.
+    error OptimismPortal_CalldataTooLarge();
+
+    /// @notice Thrown when the portal is paused.
+    error OptimismPortal_CallPaused();
+
+    /// @notice Thrown when a gas estimation transaction is being executed.
+    error OptimismPortal_GasEstimation();
+
+    /// @notice Thrown when the gas limit for a deposit is too low.
+    error OptimismPortal_GasLimitTooLow();
+
+    /// @notice Thrown when the target of a withdrawal is not a proper dispute game.
+    error OptimismPortal_ImproperDisputeGame();
+
+    /// @notice Thrown when a withdrawal has not been proven against a valid dispute game.
+    error OptimismPortal_InvalidDisputeGame();
+
+    /// @notice Thrown when a withdrawal has not been proven against a valid merkle proof.
+    error OptimismPortal_InvalidMerkleProof();
+
+    /// @notice Thrown when a withdrawal has not been proven against a valid output root proof.
+    error OptimismPortal_InvalidOutputRootProof();
+
+    /// @notice Thrown when a withdrawal's timestamp is not greater than the dispute game's creation timestamp.
+    error OptimismPortal_InvalidProofTimestamp();
+
+    /// @notice Thrown when the root claim of a dispute game is invalid.
+    error OptimismPortal_InvalidRootClaim();
+
+    /// @notice Thrown when a withdrawal is being finalized by a reentrant call.
+    error OptimismPortal_NoReentrancy();
+
+    /// @notice Thrown when a withdrawal has not been proven for long enough.
+    error OptimismPortal_ProofNotOldEnough();
+
+    /// @notice Thrown when a withdrawal has not been proven.
+    error OptimismPortal_Unproven();
+
+    /// @notice Thrown when the caller is not authorized to call the function.
+    error OptimismPortal_Unauthorized();
+
+    /// @notice Thrown when the wrong proof method is used.
+    error OptimismPortal_WrongProofMethod();
+
+    /// @notice Thrown when a super root proof is invalid.
+    error OptimismPortal_InvalidSuperRootProof();
+
+    /// @notice Thrown when an output root index is invalid.
+    error OptimismPortal_InvalidOutputRootIndex();
+
+    /// @notice Thrown when an output root chain id is invalid.
+    error OptimismPortal_InvalidOutputRootChainId();
+
+    /// @notice Thrown when trying to migrate to the same AnchorStateRegistry.
+    error OptimismPortal_MigratingToSameRegistry();
+
+    /// @notice Semantic version.
+    /// @custom:semver 4.7.0
+    function version() public pure virtual returns (string memory) {
+        return "4.7.0";
+    }
+
+    /// @param _proofMaturityDelaySeconds The proof maturity delay in seconds.
+    constructor(uint256 _proofMaturityDelaySeconds) ReinitializableBase(3) {
```

Comment:
OK, thinking out loud, just like `OptimismPortal2`, it should be possible to upgrade to this contract from either U15 or U16. Is that correct? If so then `3` is correct.